__tests__/functional/tagged-format.test.ts

```diff
@@ -0,0 +1,114 @@
+import {
+  TaggedTypeFormat,
+  DocumentReference,
+  Module,
+} from "../../src/tagged-type";
+
+describe("tagged format", () => {
+  it("can be decoded", () => {
+    const allTypes: string = `{
+      "bugs_coll": { "@mod": "Bugs" },
+      "bug": { "@doc": "Bugs:123" },
+      "name": "fir",
+      "age": { "@int": "200" },
+      "birthdate": { "@date": "1823-02-08" },
+      "molecules": { "@long": "999999999999999999" },
+      "circumference": { "@double": "3.82" },
+      "created_at": { "@time": "2003-02-08T13:28:12.000555+00:00" },
+      "extras": {
+        "nest": {
+          "@object": {
+            "num_sticks": { "@int": "58" },
+            "@extras": { "egg": { "fertilized": false } }
+          }
+        }
+      },
```

Comment:
In this part of the test, `@object` indicates a tagged object and `@extras` is literally the field name. since `@extras` is an object, then it should also include an `@object` field, as should `egg`

```suggestion
      "extras": {
        "nest": {
          "@object": {
            "num_sticks": { "@int": "58" },
            "@extras": { 
              "@object": {
                "egg": { 
                  "@object": { "fertilized": false } 
                }
              } 
            }
          }
        }
      },
```

Reply:
from my understanding `@object` is optional. from local testing it seems that nesting the contents of `egg` doesn't change the result any. the object structure was copied directly from what we're using to [validate Python driver](https://github.com/fauna/fauna-python/blob/4ac342a22aaf9d30b3175d3e80c4c7a410c69ee7/tests/fixtures/all_types.py#L5-L11) -- so if we feel strongly that we _need_ this then we should update in both places. However, I don't see that it's necessary.